### ISSUE TITLE
Formatting Bug With Custom Type When Multiple Values Are Being Formatted

### ISSUE THREAD

**Issue by USAFrenzy** [USAFrenzy/ArgFormatter#1]
ISSUE:
------------------------------------------------------------------------
- When formatting a custom type value, if you have multiple substitution brackets present using the custom type, it will format the first value encountered correctly, however, subsequent formatting ends up bypassing the custom parsing and errors on a custom specifier, such as '%' for example if '%' was being used as a flag encounter token, thinking it's parsing a native type instead of a custom type. This is most likely interrelated with #2 
-----------------------------
 EXAMPLES: 
-----------------------------
  ```cpp 
   formatter::format("{:%L} - {:%l}", someCustomVariable);
   ```
  - the above formats the first bracket correctly and errors on the second bracket
   ```cpp 
   formatter::format("{0:%L} - {0:%l}", someCustomVariable);
   ```
  - the above formats the first bracket correctly and errors on the second bracket
  ```cpp 
   formatter::format("{:%L}", someCustomVariable);
   ```
  - the above formats the argument as expected
  ```cpp 
   formatter::format("{0:%L}", someCustomVariable);
   ```
  - the above formats the argument as expected
------------------------------------------------------------------------
Initial thoughts are to:
------------------------------------------------------------------------

- Test if a very quick edit to clear the storage array and value type array fixes the issue
- If the first point doesn't fix it,  then dive into the code base and most likely end up testing an intermediate step between custom types and native types as a sort of "handshake" process

**Comment by USAFrenzy** [comment]
Fixed in commit 919b806

**Comment by USAFrenzy** [comment]
Reopening this issue since it seems to not have been fully fixed -> issue #3 denotes the case where this hasn't been fully fixed.

**Comment by USAFrenzy** [comment]
Issue fixed with [v1.0.3 release](https://github.com/USAFrenzy/ArgFormatter/releases/tag/v1.0.3)
